### ISSUE TITLE
steam closed after ~10min (>10.000 achievements) unlocking

### ISSUE THREAD

**Issue by Gummibeer** [gotkrypto76/SteamAchievementManager#3]
After it ran about 10min and unlocked 10.900 achievements steam closed itself.

**Dump Summary**

| key | value |
|---|---|
| Dump File	| crash_steam.exe_20180706213818_1.dmp |
| Last Write Time	| 06.07.2018 21:38:18 |
| Process Name	| Steam.exe  |
| Process Architecture	| x86 |
| Exception Code | 	0xC0000005 |
| Exception Information	| The thread tried to read from or write to a virtual address for which it does not have the appropriate access. |
| Heap Information	| Not Present |

**Comment by gotkrypto76** [comment]
Thanks for submitting your bug report. I have a hunch on what is causing this, and if that is in fact the case, it may be an issue with steam and the influx of achievements, not SAM Auto itself. I’d like to confirm this though, and possibly work out a fix for it when time permits. If you’re willing, I’d really appreciate if you could send the crash .dmp file mentioned.

**Comment by Gummibeer** [comment]
The whole one? The summary is above but yes I can.

I've noticed the count because I also think that the steam achievement system isn't made for 10k unlocks on "same" time.

I would think that the only solution would be a timeout based on new unlocked achievements.

**Comment by Gummibeer** [comment]
[crash_steam.exe_20180706213818_1.dmp.zip](https://github.com/gotkrypto76/SteamAchievementManager/files/2171803/crash_steam.exe_20180706213818_1.dmp.zip)

**Comment by gotkrypto76** [comment]
Yeah, I’d appreciate the whole one if you don’t mind. I’m not sure if there is any personal information in it, but you might want to skim it just to make sure it’s clean.

I’m fairly certain that you are right about the cause and fix. I’ll try and think up a good way to implement it.
